dotnet/src/Microsoft.Agents.AI.DurableTask/DurableAIAgent.cs

```diff
@@ -95,7 +99,17 @@ public override async Task<AgentRunResponse> RunAsync(
         }
 
         RunRequest request = new([.. messages], responseFormat, enableToolCalls, enableToolNames);
-        return await this._context.Entities.CallEntityAsync<AgentRunResponse>(durableThread.SessionId, nameof(AgentEntity.RunAgentAsync), request);
+        try
+        {
+            return await this._context.Entities.CallEntityAsync<AgentRunResponse>(
+                durableThread.SessionId,
+                nameof(AgentEntity.RunAgentAsync),
+                request);
+        }
+        catch (EntityOperationFailedException e) when (e.FailureDetails.ErrorType == "EntityTaskNotFound")
+        {
+            throw new AgentNotRegisteredException(this._agentName, e);
```

Comment:
Apart from a missing agent registration, are there any other reasons why CallEntityAsync might throw the `EntityOperationFailedException ` exception with that ErrorType value "EntityTaskNotFound"?